### PR TITLE
[TIMOB-19567] iOS: Deprecate useractivitywillsave event

### PR DESCRIPTION
**Pull request by cheekiatng**
JIRA: https://jira.appcelerator.org/browse/TIMOB-19567

### CODE REVIEW

File: iphone/Classes/TiAppiOSUserActivityProxy.m [hunk 1]
```diff
@@ -186,7 +186,7 @@ -(NSNumber*)isValid
 
 -(NSNumber*)supported
 {
-    DEPRECATED_REPLACED(@"App.IOS.UserActivity.getSupported()" ,@"5.1.0",@"App.IOS.UserActivity.isSupported()")
+    DEPRECATED_REPLACED(@"App.IOS.UserActivity.getSupported()" ,@"5.1.0",@"Titanium.App.IOS.UserActivity.isSupported()")
```
Comment (FokkeZB) [inline iphone/Classes/TiAppiOSUserActivityProxy.m]: Should the first line be prefixed with "Titanium" as well?

File: iphone/Classes/TiAppiOSUserActivityProxy.m [hunk 2]
```diff
@@ -197,6 +197,7 @@ -(NSNumber*)supported
 - (void)userActivityWillSave:(NSUserActivity *)userActivity
 {
     if([self _hasListeners:@"useractivitywillsave"]){
+        DebugLog(@"[WARN] Titanium.App.IOS.UserActivity.useractivitywillsave event is deprecated. Update user activity by setting Titanium.App.IOS.UserActivity.needsSave property to true instead.");
```
Comment (FokkeZB) [inline iphone/Classes/TiAppiOSUserActivityProxy.m]: Just setting needsSave is not gonna do it. I think it's better to put it this way:

> Update user activity and then set the Titanium.App.iOS.UserActivity.needsSave property to true every time you have a change.

File: apidoc/Titanium/App/iOS/UserActivity.yml [hunk 3]
```diff
@@ -85,17 +85,7 @@ properties:
...
-    description: |
-        Set the property to `true` if the user activity needs to be updated before handing it off to
-        another device, then listen for the `useractivitywillsave` event to update the state or content
-        of the user activity.
-
-        iOS will reset the property to false after the `useractivitywillsave` event fires.
-
-        Note that when the user activity first becomes active, the `useractivitywillsave` event
-        will fire once even if the property is set to false.
+        Set to true everytime the user activity needs to be updated before handing it off to another device.
```
Comment (FokkeZB) [inline apidoc/Titanium/App/iOS/UserActivity.yml]: I think this is more clear:

> Set to true everytime you have updated the user activity and need the changes to be saved before handing it off to another device.

File: iphone/Classes/TiAppiOSUserActivityProxy.m (hunk 1 above)
Comment (hansemannn) [inline iphone/Classes/TiAppiOSUserActivityProxy.m]: We have a weird behavior in the macro that I discovered last week. The prefix is added to the first part, but not to the second one. So this is right for now and will be refactored along with the other used macros in the future to unify the macro.

**Comment by hansemannn** [comment]
Thanks @FokkeZB for checking the docs. Will merge after @cheekiatng is back.

**Comment by hansemannn** [comment]
Approved!

**Pull request merged**